simulator/func_sim/alu.h

```diff
@@ -234,6 +234,13 @@ struct ALU
 
     // Bit manipulations
     template<typename I> static void sbext( I* instr) { instr->v_dst[0] = 1U & ( instr->v_src[0] >> shamt_v_src2<typename I::RegisterUInt>( instr)); }
+    
+    template<typename I> static void max( I* instr)
+    { 
+	instr->v_dst[0] = std::max( static_cast<typename I::RegisterSInt>( instr->v_src[0]), static_cast<typename I::RegisterSInt>( instr->v_src[1]));
```

Comment:
Can we use `std::max<typename I::RegisterSInt>( instr->v_src[0], instr->v_src[1])` ?

Reply:
I do not know why, but there is a problem with max (maxu works and builds fine). I just do not understand how to convert unsigned to signed and compare them. I tried using I::RegisterSInt, I::RegisterUInt and even sign_t with them but none of these work (fails when building). What is wrong with my ALU max method?

Reply:
Well, probably it means that answer to my question was "no".
Let's roll back to your original proposal, but with `narrow_cast` macro instead of `static_cast`.

Reply:
I updated opcodes_rv32b with max and maxu instructions, but they do not appear when i clone the mipt-mips repository recursively. What am I doing wrongly? I have already sent a pull request yesterday.

Reply:
You were right, I made adjustments to my local op_codes. I have sent a pull request.

Reply:
You succesfully added new instructions to submodule. No extra pull requests are needed.
However, submodule version should be updated within main Git repository.
Please use the  commands I typed below (https://github.com/MIPT-ILab/mipt-mips/pull/1387#issuecomment-721658364)